src/main/scala/firrtl/passes/RemoveAccesses.scala

```diff
@@ -101,7 +103,8 @@ object RemoveAccesses extends Pass {
                 stmts += wire
                 rs.zipWithIndex foreach {
                   case (x, i) if i < temps.size =>
-                    stmts += Connect(get_info(s),getTemp(i),x.base)
+                    stmts += IsInvalid(get_info(s),getTemp(i))
+                    stmts += Conditionally(get_info(s),x.guard,Connect(get_info(s),getTemp(i),x.base),EmptyStmt)
```

Comment:
This isn't that important but do you think maybe we should change the naming of the intermediates here?
For something like `out <= arr[idx]`
```verilog
// Rather than this
  assign _GEN_1 = 2'h1 == idx ? arr_1 : arr_0;
  assign _GEN_2 = 2'h2 == idx ? arr_2 : _GEN_1;
  assign _GEN_3 = 2'h3 == idx ? arr_3 : _GEN_2;
  assign out = _GEN_3;

// Emit this?
  assign _arr_idx_1 = 2'h1 == idx ? arr_1 : arr_0;
  assign _arr_idx_2 = 2'h2 == idx ? arr_2 : _arr_idx_1;
  assign _arr_idx_3 = 2'h3 == idx ? arr_3 : _arr_idx_2;
  assign out = _arr_idx_3;
// So long as we prefix with _, constant propping will do the right thing with out
```

Reply:
Sounds good, I pushed a fix.

Reply:
Kind of hilariously, your change gives better names but expand whens prevents it from being super good. Oh well, not that important.

```
======== Starting Transform ExpandConnects$ ========                           
----------------------------------------------------                           
                                                                                                                                               
circuit index : @[:index.fir@2.0]                                              
  module index : @[:index.fir@3.2]                                             
    input idx : UInt<2> @[:index.fir@4.4]                                      
    input vec : UInt<8>[4] @[:index.fir@5.4]                                   
    output out : UInt<8> @[:index.fir@6.4]                                     
                                                                               
    out <= vec[idx] @[:index.fir@8.4]                                          
                                                                               
======== Finished Transform ExpandConnects$ ========                           
                                                                               
======== Starting Transform RemoveAccesses$ ========                           
----------------------------------------------------                           
                                                                                                                                                  
circuit index : @[:index.fir@2.0]                                              
  module index : @[:index.fir@3.2]                                             
    input idx : UInt<2> @[:index.fir@4.4]                                      
    input vec : UInt<8>[4] @[:index.fir@5.4]                                   
    output out : UInt<8> @[:index.fir@6.4]                                     
                                                                               
    wire _vec_idx : UInt<8> @[:index.fir@8.4]                                  
    _vec_idx is invalid @[:index.fir@8.4]                                      
    when eq(UInt<1>("h0"), idx) : @[:index.fir@8.4]                            
      _vec_idx <= vec[0] @[:index.fir@8.4]                                     
    when eq(UInt<1>("h1"), idx) : @[:index.fir@8.4]                            
      _vec_idx <= vec[1] @[:index.fir@8.4]                                     
    when eq(UInt<2>("h2"), idx) : @[:index.fir@8.4]                            
      _vec_idx <= vec[2] @[:index.fir@8.4]                                     
    when eq(UInt<2>("h3"), idx) : @[:index.fir@8.4]                            
      _vec_idx <= vec[3] @[:index.fir@8.4]                                                                                                                    
    out <= _vec_idx @[:index.fir@8.4]                                          
                                                                               
======== Finished Transform RemoveAccesses$ ========                           
                                                                               
======== Starting Transform ExpandWhens$ ========                              
-------------------------------------------------                              
                                                                                                                                               
circuit index : @[:index.fir@2.0]                                              
  module index : @[:index.fir@3.2]                                             
    input idx : UInt<2> @[:index.fir@4.4]                                      
    input vec : UInt<8>[4] @[:index.fir@5.4]                                   
    output out : UInt<8> @[:index.fir@6.4]                                     
                                                                               
    wire _vec_idx : UInt<8> @[:index.fir@8.4]                                  
    node _GEN_0 = validif(eq(UInt<1>("h0"), idx), vec[0]) @[:index.fir@8.4]    
    node _GEN_1 = mux(eq(UInt<1>("h1"), idx), vec[1], _GEN_0) @[:index.fir@8.4]
    node _GEN_2 = mux(eq(UInt<2>("h2"), idx), vec[2], _GEN_1) @[:index.fir@8.4]
    node _GEN_3 = mux(eq(UInt<2>("h3"), idx), vec[3], _GEN_2) @[:index.fir@8.4]
    out <= _vec_idx                    
    _vec_idx <= _GEN_3                 

======== Finished Transform ExpandWhens$ ========        
```